File: app/src/main/java/com/bll/lnkstudy/dialog/CommonDialog.java
package com.bll.lnkstudy.dialog;

import android.app.AlertDialog;
import android.content.Context;
import android.text.TextUtils;
import android.view.ContextThemeWrapper;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.TextView;

import com.bll.lnkstudy.Constants;
import com.bll.lnkstudy.R;
import com.bll.lnkstudy.utils.DP2PX;


/**
 * Created by dell on 2017/10/23.
 */

public class CommonDialog {
    private Context context;
    private AlertDialog dialog;
    private TextView okTv;
    private String title="";
    private boolean is;
    private String content="";//提示文案
    private String cancle="取消";//取消文案
    private String ok ="确认";//确认文案
    private int screenPos=0;

    public CommonDialog(Context context) {
        this.context = context;
    }
    public CommonDialog(Context context,int screenPos) {
        this.context = context;
        this.screenPos=screenPos;
    }

    public CommonDialog setTitle(String title) {
        this.title = title;
        return this;
    }
    public CommonDialog setContent(String content) {
        this.content = content;
        return this;
    }
    public CommonDialog setCancel(String cancle) {
        this.cancle = cancle;
        return this;
    }
    public CommonDialog setOk(String ok) {
        this.ok = ok;
        return this;
    }

    public CommonDialog builder() {
        View view = LayoutInflater.from(context).inflate(R.layout.dialog_com, null);
        TextView titleTv = view.findViewById(R.id.tv_dialog_title);
        TextView contentTv = view.findViewById(R.id.tv_dialog_content);
        TextView cancleTv = view.findViewById(R.id.tv_cancle);
        okTv=view.findViewById(R.id.tv_ok);

        if(!TextUtils.isEmpty(title)) titleTv.setText(title);
        if (!is)
        {
            titleTv.setVisibility(View.GONE);
            contentTv.setMinHeight(220);
        }
        if(!TextUtils.isEmpty(content)) contentTv.setText(content);
        if(!TextUtils.isEmpty(cancle)) cancleTv.setText(cancle);
        if(!TextUtils.isEmpty(ok))okTv.setText(ok);

        cancleTv.setOnClickListener(v -> {
            cancel();
            if(onDialogClickListener !=null) onDialogClickListener.cancel();
        });
        okTv.setOnClickListener(v -> {
            cancel();
            if(onDialogClickListener !=null) onDialogClickListener.ok();
        });
        dialog = new AlertDialog.Builder(new ContextThemeWrapper(context,R.style.styleDialogCustom)).create();
        dialog.setView(view);
        dialog.show();
        Window window=dialog.getWindow();
        window.setBackgroundDrawableResource(android.R.color.transparent);
        window.getDecorView().setPadding(0, 0, 0, 0);
        WindowManager.LayoutParams lp = window.getAttributes();
        lp.width = 600;
        lp.height = WindowManager.LayoutParams.WRAP_CONTENT;

        if (screenPos==3){
            lp.gravity = Gravity.CENTER_VERTICAL | Gravity.RIGHT;
            lp.x=(Constants.Companion.getWIDTH()- 600)/2;
        }

        window.setAttributes(lp);
        return this;
    }

    public CommonDialog setTitleView(boolean is) {
        this.is=is;
        return this;
    }

    public void show() {
        dialog.show();
    }
    public void cancel() {
        dialog.dismiss();
    }
    public OnDialogClickListener onDialogClickListener;
    public interface OnDialogClickListener {
        void cancel();
        void ok();
    }
    public void setDialogClickListener(OnDialogClickListener onDialogClickListener){
        this.onDialogClickListener = onDialogClickListener;
    }
}

File: app/src/main/java/com/bll/lnkstudy/dialog/CommonDialog.kt
package com.bll.lnkstudy.dialog

import android.app.Dialog
import android.content.Context
import android.view.Gravity
import android.view.View
import android.widget.TextView
import com.bll.lnkstudy.Constants
import com.bll.lnkstudy.R
import com.bll.lnkstudy.utils.DP2PX


class CommonDialog(private val context: Context,private val screenPos:Int) {

    private var dialog: Dialog? = null
    private var titleStr = ""
    private var contentStr = "" //提示文案
    private var contentStrId = 0 //提示文案
    private var cancelStr = "取消" //取消文案
    private var okStr = "确认" //确认文案

    constructor(context: Context):this(context,0)

    fun setTitle(title: String): CommonDialog {
        this.titleStr = title
        return this
    }

    fun setContent(content: String): CommonDialog {
        this.contentStr = content
        return this
    }

    fun setContent(contentId: Int): CommonDialog {
        this.contentStrId = contentId
        return this
    }

    fun setCancel(cancel: String): CommonDialog {
        cancelStr = cancel
        return this
    }

    fun setOk(ok: String): CommonDialog {
        okStr = ok
        return this
    }

    fun builder(): CommonDialog {
        dialog = Dialog(context)
        dialog?.setContentView(R.layout.dialog_com)
        val window=dialog?.window!!
        window.setBackgroundDrawableResource(android.R.color.transparent)
        val layoutParams =window.attributes
        if (screenPos==1){
            layoutParams.gravity = Gravity.CENTER_VERTICAL or Gravity.LEFT
            layoutParams.x=(Constants.WIDTH- DP2PX.dip2px(context,450F))/2
        }
        else{
            layoutParams.gravity = Gravity.CENTER_VERTICAL or Gravity.RIGHT
            layoutParams.x=(Constants.WIDTH- DP2PX.dip2px(context,450F))/2
        }
        dialog?.show()

        val titleTv = dialog!!.findViewById<TextView>(R.id.tv_dialog_title)
        val contentTv = dialog!!.findViewById<TextView>(R.id.tv_dialog_content)
        val cancelTv = dialog!!.findViewById<TextView>(R.id.tv_cancel)
        val tvOk = dialog!!.findViewById<TextView>(R.id.tv_ok)

        if (titleStr.isNotEmpty()) titleTv.text = titleStr
        titleTv.visibility =  if (titleStr.isNotEmpty()) View.VISIBLE else View.GONE

        if (contentStr.isNotEmpty()) contentTv.text = contentStr
        if (contentStrId!=0) contentTv.setText(contentStrId)
        if (cancelStr.isNotEmpty()) cancelTv.text = cancelStr
        if (okStr.isNotEmpty()) tvOk.text = okStr

        cancelTv.setOnClickListener {
            cancel()
            onDialogClickListener?.cancel()
        }
        tvOk.setOnClickListener {
            cancel()
            onDialogClickListener?.ok()
        }

        return this
    }

    fun show() {
        dialog?.show()
    }

    fun cancel() {
        dialog?.dismiss()
    }

    var onDialogClickListener: OnDialogClickListener? = null

    interface OnDialogClickListener {
        fun cancel()
        fun ok()
    }

    fun setDialogClickListener(onDialogClickListener: OnDialogClickListener?) {
        this.onDialogClickListener = onDialogClickListener
    }
}